handler.go

```diff
@@ -236,8 +240,12 @@ func (h *Handler) handleOffsets(ctx *Context, req *protocol.OffsetsRequest) *pro
 			ts = time.Unix(0, 0)
 		} else if reqTopic.Partitions[0].Timestamp == -1 {
 			// Latest = all the data up until now
-			offset = math.MaxInt64 // Unlimited data
 			ts = time.Now()
+			if h.config.MaxOffset != nil {
+				offset = h.config.MaxOffset(ts)
```

Comment:
It appears that the only timestamp that `MaxOffset` is ever called with is `time.Now()`, so it seems like the MaxOffsetFn type could just as easily be `func() int64`, which feels a bit more generic in terms of allowing whatever offset behavior the user specifies and not emphasizing timeful behaviors quite as much.

Reply:
Good idea, I went ahead and made that update. Seems better.